core/src/main/java/org/testcontainers/utility/TestcontainersConfiguration.java

```diff
@@ -129,29 +138,29 @@ public Integer getRyukTimeout() {
 
     @Deprecated
     public String getKafkaImage() {
-        return String.valueOf(properties.getOrDefault("kafka.container.image", "confluentinc/cp-kafka"));
+        return getKafkaDockerImageName().asCanonicalNameString();
     }
 
     public DockerImageName getKafkaDockerImageName() {
-        return DockerImageName.parse(getKafkaImage());
+        return getImage("kafka.container.image", "confluentinc/cp-kafka");
     }
 
     @Deprecated
     public String getPulsarImage() {
-        return String.valueOf(properties.getOrDefault("pulsar.container.image", "apachepulsar/pulsar"));
+        return getPulsarDockerImageName().asCanonicalNameString();
     }
 
     public DockerImageName getPulsarDockerImageName() {
```

Comment:
I just realized... when did we introduce `DockerImageName getPulsarDockerImageName()`?

If 1.15.0-rc1, maybe we should use the opportunity to remove module's images from `TestcontainersConfiguration`, and instead expose something like the current `getImage` that would allow _any_ module to use the configuration mechanism?

Reply:
It was `1.15.0-rc1` so yeah, we can do that.

> that would allow any module to use the configuration mechanism?

I'd be keen to still decrease reliance on `TestcontainersConfiguration` overall. I see it as an OK default image substitutor, but in the long term I'd like module classes to not call  `TestcontainersConfiguration` at all, and instead rely entirely on the image substitution mechanism.

Reply:
Actually, I don't think we need to do this. #3102 will:

* Rip out all references from modules to `TestcontainersConfiguration` for obtaining an image name
* Make `RemoteDockerImage` use `ImageNameSubstitutor` to look up a substitute name, if set
* The default implementation of `ImageNameSubstitutor` just defers to `TestcontainersConfiguration` (so that existing uses of props-file based substitution carry on working)

If we don't do this, and essentially keep `TestcontainersConfiguration` near the top of the stack, then I think it becomes a lot harder to swap in an alternative substitutor implementation without making every module (and `GenericContainer`) use `TestcontainersConfiguration`.